src/library_pthread.js

```diff
@@ -309,7 +304,7 @@ var LibraryPThread = {
         worker.on('error', function(e) {
           worker.onerror(e);
         });
-        worker.on('detachedExit', function() {
+        worker.on('exit', function() {
```

Comment:
Why this change rather than just removing this handler completely like above?  What is the 'exit' message here?

Reply:
The `exit` -> `detachedExit` change was mistakenly introduced in PR #12861 (see comment https://github.com/emscripten-core/emscripten/pull/12861#discussion_r700102217), but perhaps it is indeed better to remove this listener completely (since issue #9763 may have already been resolved, [`test_pthread_create.cpp`](https://github.com/emscripten-core/emscripten/blob/main/tests/pthread/test_pthread_create.cpp) runs fine on Node).

Reply:
Oh I see.  Can you split that out into its own fix and mention that is address an accidental change from #12861?  Thanks for doing all this work BTW, its much appreciated cleanup.

Reply:
No problem! I've reverted this change with commit d69ff5bb431ce13a815b12989bb78fad5ae1dd0f, and will make a new PR with that (or one that removes this listener completely).